gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckExactDependenciesTask.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.tasks;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Sets;
+import com.google.common.collect.Streams;
+import com.palantir.baseline.plugins.BaselineExactDependencies;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+
+public class CheckExactDependenciesTask extends DefaultTask {
+
+    // TODO(dfox): hoist these indexes so they are not recomputed for every subproject
+    private final Map<String, ResolvedArtifact> classToDependency = new HashMap<>();
+    private final Map<ResolvedArtifact, Set<String>> classesFromArtifact = new HashMap<>();
+    private final Map<ResolvedArtifact, ResolvedDependency> artifactsFromDependency = new HashMap<>();
+
+    private final ListProperty<Configuration> dependenciesConfigurations;
```

Comment:
```suggestion
    private final ConfigurableFileCollection dependenciesConfigurations = getProject().getLayout().configurableFiles();
```

Reply:
I actually meant this for the line below..

Reply:
I don't want to accidentally take a super high gradle dependency here... I think getLayout().configurableFiles() is a new 5.0 feature...